### PR TITLE
Add Java crypto callbacks

### PR DESCRIPTION
**Pull request by jyemin**
Opening this against Kevin's crypto hook callback branch in anticipation of that branch being pushed without any further API changes.

https://jira.mongodb.org/browse/CDRIVER-3229

https://evergreen.mongodb.com/version/5d2cf96ce3c33125ddbe00da

### CODE REVIEW

File: bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CipherCallback.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private final int mode;
+
+    CipherCallback(final String algorithm, final String transformation, final int mode) {
+        this.algorithm = algorithm;
+        this.transformation = transformation;
+        this.mode = mode;
+    }
+
+    @Override
+    public boolean crypt(final Pointer ctx, final CAPI.mongocrypt_binary_t key, final CAPI.mongocrypt_binary_t iv,
```
Comment (rozza) [inline bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CipherCallback.java]: Any reason its `CAPI.mongocrypt_binary_t` and not just `monogcrypt_binary_t`?  Elsewhere in the lib the shorter form seems to be preferred

Comment (jyemin) [reply bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CipherCallback.java]: Ah, it happened when I did an auto-refactoring to move the callbacks out of MongoCryptImpl.java.  Fixed it in all of the callbacks.

File: bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPIHelper.java [hunk 2]
```diff
@@ -75,6 +75,18 @@ static ByteBuffer toByteBuffer(final mongocrypt_binary_t binary) {
...
+    static byte[] toByteArray(final mongocrypt_binary_t binary) {
+        ByteBuffer byteBuffer = toByteBuffer(binary);
+        byte[] byteArray = new byte[byteBuffer.remaining()];
+        byteBuffer.get(byteArray);
+        return byteArray;
+    }
+
+    static void writeByteArrayToBinary(final mongocrypt_binary_t binary, byte[] bytes) {
+        Pointer outPointer = mongocrypt_binary_data(binary);
+        outPointer.write(0, bytes, 0, bytes.length);
```
Comment (markbenvenuto) [inline bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPIHelper.java]: Check the mongocrypt_binary_t is large enough for bytes[]

Comment (jyemin) [reply bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPIHelper.java]: Done

Comment (jyemin) [reply bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPIHelper.java]: If this check fails it indicates a bug in libmongocrypt.  But better to report an error in that case than corrupt memory.

**Pull request closed without merging**